packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -279,6 +284,8 @@ class _EditableTextWidget extends LeafRenderObjectWidget {
     }
 
     String text = value.text;
+    if (hideText)
+      text = new String.fromCharCodes(new List<int>.filled(text.length, 0x2022));
```

Comment:
This still leaks the number of characters. We could be slightly better by just drawing a rounded rectangle of the length that the text would be, or something.